sources/PTYSession.m

```diff
@@ -8570,6 +8584,28 @@ - (BOOL)highlightCursorLine {
     return _textview.highlightCursorLine;
 }
 
+- (void)screenCursorDidMoveToCol:(int)col {
+    if (_textview.cursorVisible) {
```

Comment:
Only redraw if the vertical guide is on

Reply:
Given that this is true any time we call [...textView setNeedsDisplayOnColumn...] I have placed the check there so that most parts of the code don't need to be concerned with this implementation detail (ie the inefficiency of setNeedsDisplayOnCol), but all benefit from this check.